File: src/main/java/com/efedotov/vaultly/dto/auth/AuthResponse.java
package com.efedotov.vaultly.dto.auth;

import lombok.Data;
import java.util.Set;
import java.util.UUID;

@Data
public class AuthResponse {
    private String accessToken;
    private String refreshToken;
    private UUID userId;
    private String email;
    private String username;
    private Long storageUsed;
    private Long storageLimit;
    private Set<String> roles;
    private Boolean totpEnabled;
}

File: lib/server/model/auth/auth_response/auth_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'auth_response.freezed.dart';
part 'auth_response.g.dart';

@freezed
abstract class AuthResponse with _$AuthResponse {
  const factory AuthResponse({
    String? accessToken,
    String? refreshToken,
    required String userId,
    String? email,
    required String username,
    required int storageUsed,
    required int storageLimit,
    required Set<String> roles,
    bool? totpEnabled,
  }) = _AuthResponse;

  factory AuthResponse.fromJson(Map<String, dynamic> json) =>
      _$AuthResponseFromJson(json);
}
